source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -112,7 +117,20 @@ void MGTransferMatrixFree<dim,VectorType>::build
                                                    dirichlet_indices,
                                                    weights_unvectorized,
                                                    this->copy_indices_global_mine,
-                                                   this->ghosted_level_vector);
```

Comment:
@kronbichler may I ask you if I interprete `internal::MGTransfer::setup_transfer` correct? It looks like you pass `this->ghosted_level_vector` with two purposes: (i) initialize it on each level; (ii) use it internally to set up `weights_on_refined`.

I am trying to extend this to do-the-same-thing for each block of the block vector and, interestingly enough, this single commit breaks multigrid preconditioner tests! :-( 

In preparation for block vectors, for now I introduce a temporary always-non-block vector to be used in `internal::MGTransfer::setup_transfer`. And then I setup the actual `this->ghosted_level_vector` outside of `internal::MGTransfer::setup_transfer`. But that commit still breaks `matrix_free/parallel_multigrid_adaptive_0[2,3,4]` tests. Am I misunderstanding the role of `this->ghosted_level_vector` here? I am quite puzzled here...